src/gologin-api.js

```diff
@@ -0,0 +1,82 @@
+import puppeteer from 'puppeteer-core';
+import GoLogin from './gologin.js';
+
+export function getDefaultParams() {
+  return {
+    token: process.env.GL_API_TOKEN,
+    profile_id: process.env.GL_PROFILE_ID,
+    executablePath: process.env.GL_EXECUTABLE_PATH,
+  };
+}
+
+const createLegacyGologin = (params) => {
+  const defaults = getDefaultParams();
+  return new GoLogin({
+    ...defaults,
+    ...params,
+  });
+};
+
+export const delay = (ms = 250) => new Promise((res) => setTimeout(res, ms));
+
+export function GologinApi({ token }) {
+  if (!token) {
+    throw new Error('GoLogin API token is missing');
+  }
+  const browsers = [];
+  const legacyGls = [];
+
+  const launchLocal = async (params) => {
+    console.log('launchExistingProfile', params);
+    const legacyGologin = createLegacyGologin(params);
+    const started = await legacyGologin.startLocal();
+    console.debug({ started });
+    const browser = await puppeteer.connect({
+      browserWSEndpoint: started.wsUrl,
+      ignoreHTTPSErrors: true,
+    });
+    browsers.push(browser);
+    legacyGls.push(legacyGologin);
+    return { browser };
+  };
+
+  const launchCloudProfile = async (params) => {
```

Comment:
this function also can be moved out of `GologinApi` class
it depends on `browsers` and `token` variables
`browsers` should be available from the highest scope of this module, but `token` is local for `GologinApi` class

so to move function out we need to extend its arguments

```
const launchCloudProfile = async (token, params) => {
```

, and pass the token on invocation:

`await launchCloudProfile(this.token, params);`